From 2f4d1d5d63cd2c92cc463884d077470d5593fb58 Mon Sep 17 00:00:00 2001
From: Alexander Koskovich <zvnexus@outlook.com>
Date: Sat, 10 Jul 2021 19:47:56 -0700
Subject: [PATCH] interfaces: Introduce the QTI FM HAL.

 * Based on observing FM HCI expectations.

PugzAreCute <me@pugzarecute.com>: Fix merge conflicts when applying to PixelOS QPR2

Change-Id: I54685ff840e8c4c4e5b0de3cc1bec3eb1ed29579
Signed-off-by: Adithya R <gh0strider.2k18.reborn@gmail.com>
Signed-off-by: chrisl7 <wandersonrodriguesf1@gmail.com>
---
 Android.bp                 |  5 +++++
 fm/1.0/Android.bp          | 16 ++++++++++++++++
 fm/1.0/IFmHci.hal          | 13 +++++++++++++
 fm/1.0/IFmHciCallbacks.hal |  6 ++++++
 fm/1.0/types.hal           | 10 ++++++++++
 5 files changed, 50 insertions(+)
 create mode 100644 fm/1.0/Android.bp
 create mode 100644 fm/1.0/IFmHci.hal
 create mode 100644 fm/1.0/IFmHciCallbacks.hal
 create mode 100644 fm/1.0/types.hal

diff --git a/Android.bp b/Android.bp
index 7aef46b..6c38d43 100644
--- a/Android.bp
+++ b/Android.bp
@@ -1,3 +1,8 @@
 subdirs = [
     "*"
 ]
+
+hidl_package_root {
+    name: "vendor.qti.hardware.fm",
+    path: "vendor/qcom/opensource/interfaces/fm",
+}
\ No newline at end of file
diff --git a/fm/1.0/Android.bp b/fm/1.0/Android.bp
new file mode 100644
index 0000000..7f1935b
--- /dev/null
+++ b/fm/1.0/Android.bp
@@ -0,0 +1,16 @@
+// This file is autogenerated by hidl-gen -Landroidbp.
+
+hidl_interface {
+    name: "vendor.qti.hardware.fm@1.0",
+    root: "vendor.qti.hardware.fm",
+    system_ext_specific: true,
+    srcs: [
+        "types.hal",
+        "IFmHci.hal",
+        "IFmHciCallbacks.hal",
+    ],
+    interfaces: [
+        "android.hidl.base@1.0",
+    ],
+    gen_java: true,
+}
diff --git a/fm/1.0/IFmHci.hal b/fm/1.0/IFmHci.hal
new file mode 100644
index 0000000..4d6df1d
--- /dev/null
+++ b/fm/1.0/IFmHci.hal
@@ -0,0 +1,13 @@
+package vendor.qti.hardware.fm@1.0;
+
+import IFmHciCallbacks;
+
+interface IFmHci {
+    @entry
+    @callflow(next={"sendHciCommand", "close"})
+    initialize(IFmHciCallbacks callback);
+    @callflow(next={"sendHciCommand", "close"})
+    sendHciCommand(HciPacket command);
+    @exit
+    close();
+};
diff --git a/fm/1.0/IFmHciCallbacks.hal b/fm/1.0/IFmHciCallbacks.hal
new file mode 100644
index 0000000..58db0df
--- /dev/null
+++ b/fm/1.0/IFmHciCallbacks.hal
@@ -0,0 +1,6 @@
+package vendor.qti.hardware.fm@1.0;
+
+interface IFmHciCallbacks {
+    initializationComplete(Status status);
+    hciEventReceived(HciPacket event);
+};
diff --git a/fm/1.0/types.hal b/fm/1.0/types.hal
new file mode 100644
index 0000000..b15f7b0
--- /dev/null
+++ b/fm/1.0/types.hal
@@ -0,0 +1,10 @@
+package vendor.qti.hardware.fm@1.0;
+
+enum Status : int32_t {
+  SUCCESS,
+  TRANSPORT_ERROR,
+  INITIALIZATION_ERROR,
+  UNKNOWN
+};
+
+typedef vec<uint8_t> HciPacket;
-- 
2.44.0

